src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -1217,14 +1218,21 @@ public void UpdateBubbleContent()
             {
                 DynamoViewModel.UIDispatcher.Invoke(() =>
                 {
-                    ErrorBubble.NodeMessages.Add(new InfoBubbleDataPacket(style, topLeft, botRight, content, connectingDirection));
+                    var messageNode = new InfoBubbleDataPacket(style, topLeft, botRight, content, connectingDirection);
+                    if(ErrorBubble.NodeMessages.Count(x=> x.TextMessageExists(content)) == 0)
```

Comment:
so two thoughts here... 
1. This will stop duplicate messages from ever being added - I don't think this is the intended behavior - for example, what if we replicate over a list of inputs, we should get multiple error messages, some of which will be the same - please confirm @QilongTang @Amoursol 
2. This seems inefficient, you're checking all the messages to make sure you they are not the same, but you can bail as soon as you see any - thats if I am incorrect on point 1.

Reply:
Comment had via Slack and now resolved with clearer AC's :) Thanks for the feedback @mjkkirschner !

Reply:
@mjkkirschner @Amoursol AC updated for Jira and additional code changes needed for this PR. Currently, warning/error can only be linked with node but not port so AC could be refined more after more investigations done by @filipeotero

Reply:
This is not a comment about this PR per se but to @mjkkirschner's point I don't think that we ever got multiple error messages for a node in the case of replication, we always reported the first one. I'm not sure if that behavior was supposed to change after visual refresh but it looks to me it didn't.

Reply:
About this PR, I'm not sure I understand why `UpdateBubbleContent` is being called for nodes that are not taking part in the evaluation in the first place. For example, if a node is copy-pasted or a new node is added, unless the node in question is participating in the incremental execution of the graph, why should its bubble contents even need to be updated. It seems wasteful. I would expect this function to be called only for the new or modified nodes or the nodes downstream of them.

Reply:
I think in line [1195](https://github.com/DynamoDS/Dynamo/pull/12653/files#diff-97f7c391a3f292149633414ed4ac50ce1a73a1439bcc3307d20da0fb043a6d78R1195) we are already checking if the node is not involved in the execution but not sure why are we also checking if the node does **not** have an error/warning state.

Reply:
@pinzart90 do you remember why you added the `!hasErrorOrWarning` condition [here](https://github.com/DynamoDS/Dynamo/blob/e504bebb2e30659ca7881261a7ffbe5efe9f4269/src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs#L1194)?
It looks like we should be able to simply skip updating the bubble content (adding new warnings) for nodes not involved in an execution.

Reply:
This is the PR that changed that https://github.com/DynamoDS/Dynamo/pull/12346

Reply:
Hmmm I see, TY @pinzart90